shared/src/main/java/com/couchbase/lite/FileLogger.java

```diff
@@ -0,0 +1,159 @@
+//
+// FileLogger.java
+//
+// Copyright (c) 2018 Couchbase, Inc All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+package com.couchbase.lite;
+
+import com.couchbase.litecore.C4Constants;
+import com.couchbase.litecore.C4Log;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+public final class FileLogger implements Logger {
+    private LogLevel _level = LogLevel.INFO;
+    private String _directory;
+    private boolean _customDirectory;
+    private boolean _hasConfigChanges;
+    private int _maxRotateCount = 1;
+    private long _maxSize = 1024;
+    private boolean _usePlaintext;
+    private final HashMap<LogDomain, String> _domainObjects = new HashMap<>();
+    private static String DefaultDirectory;
+
+    FileLogger() {
+        setupDomainObjects();
+    }
+
+    public int getMaxRotateCount() {
+        return _maxRotateCount;
+    }
+
+    public void setMaxRotateCount(int maxRotateCount) {
+        if(_maxRotateCount == maxRotateCount) {
+            return;
+        }
+
+        _maxRotateCount = maxRotateCount;
+        _hasConfigChanges = true;
+    }
+
+    public long getMaxSize() {
+        return _maxSize;
+    }
+
+    public void setMaxSize(long maxSize) {
+        if(_maxSize == maxSize) {
+            return;
+        }
+
+        _maxSize = maxSize;
+        _hasConfigChanges = true;
+    }
+
+    public boolean getUsePlaintext() {
+        return _usePlaintext;
+    }
+
+    public void setUsePlaintext(boolean usePlaintext) {
+        if(_usePlaintext == usePlaintext) {
+            return;
+        }
+
+        _usePlaintext = usePlaintext;
+        _hasConfigChanges = true;
+    }
+
+
+    public String getDirectory() {
+        return _directory;
+    }
+
+    public void setDirectory(String directory) {
+        if(directory == null) {
```

Comment:
Wondering if we should just guard null here?

Reply:
Oh I see here, I think we don't allow to set null, users can't set it back to default. I think I need to change iOS implementation for this.